ephios/event_management/signup/disposition.py

```diff
@@ -0,0 +1,167 @@
+    This also includes users that already have a participation, as that might have gotten removed in JS.
+
+    This also includes users that can normally not see the event. The permission will be added accordingly.
+    If needed, this method could be moved to signup methods.
+    """
+    return UserProfile.objects.all()
+
+
+class AddUserForm(forms.Form):
+    user = forms.ModelChoiceField(
+        widget=ModelSelect2Widget(
+            model=UserProfile,
+            search_fields=["first_name__icontains", "last_name__icontains"],
+            attrs={"form": "add-user-form", "data-placeholder": _("search")},
+        ),
+        queryset=UserProfile.objects.none(),  # set using __init__
+    )
+    new_index = forms.IntegerField(widget=forms.HiddenInput)
+
+    def __init__(self, user_queryset, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["user"].queryset = user_queryset
+
+
+class DispositionBaseViewMixin(CustomPermissionRequiredMixin, SingleObjectMixin):
+    permission_required = "event_management.change_event"
+    model = Shift
+
+    def setup(self, request, *args, **kwargs):
+        super().setup(request, *args, **kwargs)
+        self.object: Shift = self.get_object()
+
+    def dispatch(self, request, *args, **kwargs):
+        if self.object.signup_method.disposition_participation_form_class is None:
+            raise Http404(_("This signup method does not support disposition."))
+        return super().dispatch(request, *args, **kwargs)
+
+    def get_permission_object(self):
+        return self.object.event
+
+
+class AddUserView(DispositionBaseViewMixin, TemplateResponseMixin, View):
+    def get_template_names(self):
+        return [
+            self.object.signup_method.disposition_participation_form_class.disposition_participation_template
+        ]
+
+    def post(self, request, *args, **kwargs):
+        shift = self.object
+        form = AddUserForm(
+            data=request.POST,
+            user_queryset=addable_users(shift),
+        )
+        if form.is_valid():
+            user: UserProfile = form.cleaned_data["user"]
+            instance = shift.signup_method.get_participation_for(user.as_participant())
+            instance.state = AbstractParticipation.States.RESPONSIBLE_ADDED
+            instance.save()
+
+            DispositionParticipationFormset = get_disposition_formset(
+                self.object.signup_method.disposition_participation_form_class
+            )
+            formset = DispositionParticipationFormset(
+                queryset=AbstractParticipation.objects.filter(pk=instance.pk),
+                prefix="participations",
+                start_index=form.cleaned_data["new_index"],
+            )
+            form = next(filter(lambda form: form.instance.id == instance.id, formset))
+            return self.render_to_response({"form": form})
+        raise Http404("User does not exist")
```

Comment:
translation?

Reply:
this is not displayed, could remove the string entirely